src/lib/confmgt/typedvar.c

```diff
@@ -75,7 +76,15 @@ typed_var_kvassign(void *target, const config_line_t *line,
     return def->fns->kv_parse(target, line, errmsg, def->params);
   }
 
-  return typed_var_assign(target, line->value, errmsg, def);
+  int rv = typed_var_assign(target, line->value, errmsg, def);
+  if (rv < 0 && *errmsg != NULL) {
+    /* typed_var_assign() didn't know the line's keyword, but we do.
+     * Let's add it to the error message. */
```

Comment:
This seems like it's working around a design limitation in the existing code. I'm afraid I don't have a good alternative at the moment, though.